pyuvdata/parameter.py

```diff
@@ -302,7 +302,7 @@ def __eq__(self, other, silent=False):
             When set to False (default), descriptive text is printed out when parameters
             do not match. If set to True, this text is not printed.
         """
-        if isinstance(other, self.__class__):
+        if isinstance(other, self.__class__) and isinstance(self, other.__class__):
```

Comment:
is the point here to require their classes to be exactly the same? The first `isinstance` requires `other` to be either the same class or a subclass, and the second `isinstance` turns that around, which I think means that `self.__class__ == other.__class__` would give the same thing?

Reply:
I added this because there was a case where the first one returned as True when I did not expect it, so I was getting a weird error downstream that I traced to here. I don't quite remember what all the details are, I can try to figure it out.

Reply:
Right, I remember. If `other` is a subclass of `self` this statement returned as `True` before this modification. There's now a test that errors without this modification.

Reply:
That's correct. Often you want that (i.e. if it's a subclass and matches other criteria, it's fine to call it equal). Maybe here you don't though :-)